src/hyperloglog.c

```diff
@@ -418,7 +411,6 @@ uint64_t MurmurHash64A (const void * key, int len, unsigned int seed) {
         k |= (uint64_t) data[5] << 40;
         k |= (uint64_t) data[6] << 48;
         k |= (uint64_t) data[7] << 56;
-#endif
```

Comment:
I'm a bit confused here. Bit shift version looks same to me with pointer load or mempcy even for little endian architectures.

Reply:
that's exactly the point i wanted to argue, we can write simpler code (or in this case actually more complex), and assume (or even verify that) a modern compiler will do the right thing, but i rather not assume these (at least not in places where it really matters, i.e. performance intense loops).
i would rather keep redis efficient (and correct) even if compiled on an odd / old compiler.